angular-hometask/src/app/core/breadcrumbs/breadcrumbs.component.ts

```diff
@@ -1,15 +1,62 @@
 import { Component, OnInit } from '@angular/core';
+import { Router, NavigationEnd, ActivatedRoute, PRIMARY_OUTLET, Params, UrlSegment } from '@angular/router';
+interface Breadcrumb {
+  label: string;
+  params: Params;
+  url: string;
+}
 
 @Component({
   selector: 'app-breadcrumbs',
   templateUrl: './breadcrumbs.component.html',
   styleUrls: ['./breadcrumbs.component.scss']
 })
 export class BreadcrumbsComponent implements OnInit {
-
-  constructor() { }
+  public breadcrumbs: Breadcrumb[] = [];
+  constructor(private router: Router, private activatedRoute: ActivatedRoute) { }
 
   ngOnInit() {
```

Comment:
Really good behavior it add modificator of access and return type, it for all functions

Reply:
Done. I hope I didn't miss anything. 8d84c225b3ec6bd336026ab378dc97fdbe1ebee9